### PR TITLE
feat(search):add feature search adapter support

### PR DESCRIPTION
**Pull request by mahdi00021**
please check in new PR

### CODE REVIEW

**Comment by andersevenrud** [comment]
Refer to the review I made in #202 . All of what I wrote there is still an issue here.

**Comment by andersevenrud** [comment]
I'll make a new review just to make it easier... hold on...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog for osjs-client
```
Comment (andersevenrud) [inline CHANGELOG.md:1]: This file should not have changes.

File: __tests__/search.js [hunk 2]
```diff
@@ -35,19 +35,77 @@ class MockVFSServiceProvider extends ServiceProvider {
...
+                    mtime: "1985-10-26T08:15:00.000Z",
+                    mtimeMs: 499162500000,
+                    nlink: 1,
+                    rdev: 0,
+                    size: 1959,
+                    uid: 0
+                }
+            }
+        ];
+        return Promise.all(obj)
```
Comment (andersevenrud) [inline __tests__/search.js:86]: There are no promises here. You can just return your obj.

File: __tests__/search.js [hunk 3]
```diff
@@ -35,19 +35,77 @@ class MockVFSServiceProvider extends ServiceProvider {
...
+    }
+
+    destroy() {
+    }
+
+    async init() {
+    }
+
+    async search(pattern) {
+        const obj = [
```
Comment (andersevenrud) [inline __tests__/search.js:51]: Call this something a bit more descriptive, i.e. `list`.

File: __tests__/search.js [hunk 4]
```diff
@@ -35,19 +35,77 @@ class MockVFSServiceProvider extends ServiceProvider {
...
+            .then(lists => [].concat(...lists));
+    }
+}
 describe('Search', () => {
   let core;
   let search;
-
+  let options = {};
+  options = {
+        args: {
```
Comment (andersevenrud) [inline __tests__/search.js:95]: There is no `args` in the API. As I wrote in the previous review *you* made this `options = { adapters: [] }`

File: __tests__/utils/locale.js [hunk 5]
```diff
@@ -75,7 +75,7 @@ describe('Locale Utils', () => {
       'en_EN',
```
Comment (andersevenrud) [inline __tests__/utils/locale.js:75]: This file should not have changes.

File: __tests__/vfs.js [hunk 6]
```diff
@@ -29,15 +29,13 @@ const otherMount = {
   }
```
Comment (andersevenrud) [inline __tests__/vfs.js:29]: This file should not have changes.

File: __tests__/window.js [hunk 7]
```diff
@@ -226,12 +226,12 @@ describe('Window', () => {
```
Comment (andersevenrud) [inline __tests__/window.js:226]: This file should not have changes.

File: package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (andersevenrud) [inline package.json:1]: This file should not have changes.

File: __tests__/search.js [hunk 9]
```diff
@@ -35,19 +35,77 @@ class MockVFSServiceProvider extends ServiceProvider {
     ];
   }
 }
+class MySearchAdapter {
```
Comment (andersevenrud) [inline __tests__/search.js:38]: The results from this adapter has to be *actually tested*. See https://github.com/os-js/osjs-client/blob/d6781762ce7948a5a9c19e16d1917f715b38de82/__tests__/search.js#L53

**Review by andersevenrud — CHANGES_REQUESTED** [review]
Looks like you commited stuff that does not belong in this PR again.

I added the review from #202 here as well.

File: CHANGELOG.md (hunk 1 above)
Comment (mahdi00021) [reply CHANGELOG.md:1]: I didn't change it, you changed it to create a search test, you told me to copy it, it was added, I didn't change it

File: __tests__/utils/locale.js (hunk 5 above)
Comment (mahdi00021) [reply __tests__/utils/locale.js:75]: > This file should not have changes.



I didn't change it, you changed it to create a search test, you told me to rebase it, it was added, I didn't change it

File: __tests__/vfs.js (hunk 6 above)
Comment (mahdi00021) [reply __tests__/vfs.js:29]: > This file should not have changes.



I didn't change it, you changed it to create a search test, you told me to rebase it, it was added, I didn't change it

File: __tests__/search.js (hunk 9 above)
Comment (mahdi00021) [reply __tests__/search.js:38]: > 

I tested It was ok

Comment (mahdi00021) [reply __tests__/search.js:38]: > The results from this adapter has to be _actually tested_. See
> 
> https://github.com/os-js/osjs-client/blob/d6781762ce7948a5a9c19e16d1917f715b38de82/__tests__/search.js#L53

I tested It was ok

File: CHANGELOG.md (hunk 1 above)
Comment (andersevenrud) [reply CHANGELOG.md:1]: But this PR should **not contain this**. It's already in master.

File: __tests__/search.js (hunk 9 above)
Comment (andersevenrud) [reply __tests__/search.js:38]: But you're not testing **anything**.

File: __tests__/utils/locale.js (hunk 5 above)
Comment (andersevenrud) [reply __tests__/utils/locale.js:75]: But this PR should **not contain this**. It's already in master. This is stuff you commited.

File: __tests__/vfs.js (hunk 6 above)
Comment (andersevenrud) [reply __tests__/vfs.js:29]: But this PR should **not contain this**. It's already in master. This is stuff you commited.

**Comment by andersevenrud** [comment]
I'm fairly sure you've done something wrong in the rebasing somewhere.

Because a **successful rebase** does not add changes that already exists in the upstream master branch.

**Comment by mahdi00021** [comment]
> 



> I'm fairly sure you've done something wrong in the rebasing somewhere.
> 
> Because a **successful rebase** does not add changes/stuff that already exists in the upstream master branch.

What should I do now to fix it?

**Comment by andersevenrud** [comment]
The easiest way if you don't know how to work with git is pretty simple; start over:

Create a new fork from the upstream repo, then a new branch, then copy **only the changes required** and make a new PR.

**Comment by mahdi00021** [comment]
> The easiest way if you don't know how to work with git is pretty simple; start over:
> 
> Create a new fork from the upstream repo, then a new branch, then copy **only the changes required** and make a new PR.

There are so many things messed up that I'm confused, I have to fork from the beginning

**Pull request closed without merging**